partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -487,10 +487,77 @@ internal object RexConverter {
             // Args
             val args = node.args.map { visitExprCoerce(it, context) }
             // Rex
+            if (node.setq != null) {
+                if (isCollAgg(node)) {
+                    return callToCollAgg(id, node.setq!!, args)
+                } else {
+                    error("Currently, only COLL_<AGG> may use set quantifiers.")
+                }
+            }
             val op = rexOpCallUnresolved(id, args)
             return rex(type, op)
         }
 
+        /**
+         * @return whether call is `COLL_<AGG>`.
+         */
+        private fun isCollAgg(node: Expr.Call): Boolean {
+            val id = node.function as? org.partiql.ast.Identifier.Symbol ?: return false
+            return id.symbol.lowercase().startsWith("coll_")
```

Comment:
Seems a bit fragile. Consider if there was a UDF starting with `coll_`.

Reply:
Added a more in-depth check.